File: API/Helpers/GameParams.cs
namespace API.Helpers
{
    public class GameParams : PaginationParams
    {
        public string SortBy { get; set; } = "Newly Added";

        public bool ReverseOrder { get; set; }

        public string PublishersFilter { get; set; }

        public string GenresFilter { get; set; }

        public int MinDownloads { get; set; }

        public int MinRating { get; set; }

        public DateTime MinReleaseDateFilter { get; set; }  = new DateTime(2000);

        public DateTime MaxReleaseDateFilter { get; set; } = DateTime.Now.AddYears(1);

        public DateTime MinDateAddedFilter { get; set; } = new DateTime(2022);

        public DateTime MaxDateAddedFilter { get; set; } = DateTime.Now;

        public string SearchString { get; set; }
    }
}

File: clientel/src/app/_models/GameParams.ts
import { addDays } from "../_services/typeHelpers";

export class GameParams {
    public constructor(init?: Partial<GameParams>) {
        Object.assign(this, init)
    }

    pageNumber: number = 1;
    pageSize: number = 6;
    sortBy: string = "Newly Added";
    reverseOrder: boolean  = false;
    publishersFilter: string[] = [];
    genresFilter: string[] = [];
    minDownloads: number = 0;
    minRating: number = 0;
    minReleaseDateFilter: Date = new Date(2000, 1, 1);
    maxReleaseDateFilter: Date = new Date(new Date().getFullYear() + 1, new Date().getMonth(), new Date().getDate());
    minDateAddedFilter: Date = new Date(2022, 9, 1);
    maxDateAddedFilter: Date = addDays(new Date(), 1);
    searchString: string;
}